lib/src/assists/flutter/wrap_with_value_listenable_builder.dart

```diff
@@ -0,0 +1,66 @@
+import 'package:analyzer/dart/element/type.dart';
+import 'package:analyzer/source/source_range.dart';
+import 'package:analyzer_plugin/utilities/range_factory.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+
+import '../../utils/ast_node_extensions.dart';
+import '../../utils/pubspec_extensions.dart';
+import '../../utils/type_checker.dart';
+
+class WrapWithValueListenableBuilder extends DartAssist {
+  @override
+  Future<void> run(
+    CustomLintResolver resolver,
+    ChangeReporter reporter,
+    CustomLintContext context,
+    SourceRange target,
+  ) async {
+    if (!context.pubspec.isFlutterProject) return;
+
+    context.registry.addInstanceCreationExpression((node) {
+      final sourceRange = switch (node.keyword) {
+        null => node.constructorName.sourceRange,
+        final keyword => range.startEnd(
+            keyword,
+            node.constructorName,
+          ),
+      };
+      if (!sourceRange.covers(target)) return;
+
+      final type = node.staticType;
+      if (type == null || !widgetChecker.isSuperTypeOf(type)) return;
+
+      final changeBuilder = reporter.createChangeBuilder(
+        message: 'Wrap with ValueListenableBuilder',
+        priority: 29,
+      );
+
+      final listenableVariable = node.valueNotifierIdentifier;
```

Comment:
Would it be better if this assist is only available on an `InstanceCreationExpression` that has an argument assigned to a `ValueNotifier`?

We could check with `InstanceCreationExpression.argumentList.arguments`.